lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1916,272 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
+        return True
+    except Exception as e:
+        log.error(f"Failed check existance of cert:'{cert_name}': {e}")
+        return False
+
+
+def delete_certificate(
+    platform: "AzurePlatform",
+    vault_url: str,
+    cert_name: str,
+    log: Logger,
+) -> bool:
+    certificate_client = get_certificate_client(vault_url, platform)
+
+    try:
+        certificate_client.begin_delete_certificate(cert_name)
+        log.info(f"Certificate {cert_name} deleted successfully.")
+        return True
+    except Exception as e:
+        log.error(f"Error deleting certificate: {e}")
+        raise LisaException(str(e))
+
+
+def rotate_certificates(
+    log: Logger,
+    vault_url: str,
+    platform: "AzurePlatform",
+    cert_name_to_rotate: str,
+) -> None:
+    # Use the helper function to get certificate_client
+    certificate_client = get_certificate_client(vault_url, platform)
+
+    try:
+        # Retrieve the old version of the certificate
+        certificate_client.get_certificate(cert_name_to_rotate)
+    except Exception as e:
+        log.error(
+            "Failed to retrieve old version of certificate:"
+            f"{cert_name_to_rotate}': {e}"
+        )
+        raise LisaException(str(e))
+
+    cert_policy = CertificatePolicy.get_default()
+
+    # Create only the specified certificate
+    for attempt in range(2):
+        try:
+            # Create certificate
+            create_certificate_poller = certificate_client.begin_create_certificate(
+                cert_name_to_rotate, policy=cert_policy
+            )
+            create_certificate_result = create_certificate_poller.result()
+            # Handle possible None value
+            if (
+                isinstance(create_certificate_result, KeyVaultCertificate)
+                and hasattr(create_certificate_result, "properties")
+                and create_certificate_result.properties
+            ):
+                new_certificate_version = create_certificate_result.properties.version
+                log.info(
+                    f"New version of certificate '{cert_name_to_rotate}': "
+                    f"{new_certificate_version}"
+                )
+                log.info("Certificate rotated")
+            else:
+                log.error(
+                    "Failed to retrieve properties from create certificate result."
+                )
+
+            certificate_client.update_certificate_properties(
+                certificate_name=cert_name_to_rotate, enabled=True
+            )
+            break
+        except Exception as e:
+            if attempt < 1:
+                sleep(1)
+            else:
+                log.error(
+                    f"Failed to rotate certificate:"
+                    f"{cert_name_to_rotate}' after 2 attempts."
+                    f"More exception info: {e}"
+                )
+                raise LisaException(str(e))
+
+
+def check_system_status(node: Node, log: Logger) -> None:
```

Comment:
This method is used by akv test case only, so move it to the test case, and as a private method (start with `_`).